components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java

```diff
@@ -123,6 +130,14 @@ public boolean validateScope(AccessTokenDO accessTokenDO, String resource) throw
             return false;
         }
 
+        // If GrantType is SAML20_BEAERER and SAML2_ASSERTION_ENABLED system property is set to true
+        // Avoid validating user roles
+        String isSAML2Enabled = System.getProperty(SAML2_ASSERTION_ENABLED);
```

Comment:
Where does this System property get set?
Can there be a case where two different entities set this property concurrently that will cause unexpected behaviours?